satpy/readers/ahi_gridded.py

```diff
@@ -0,0 +1,260 @@
+            ftp.retrbinary("RETR " + AHI_REMOTE_LUTS[2], _fp.write)
+
+    @staticmethod
+    def _untar_luts(tarred_file, outdir):
+        """Downloaded LUTs are a compressed tarball, uncompress here."""
+        import tarfile
+        tar = tarfile.open(tarred_file)
+        tar.extractall(outdir)
+        tar.close()
+        os.remove(tarred_file)
+
+    def _get_luts(self):
+        """LUTs are needed for count->REFL/BT conversion. Download them."""
+        import tempfile
+        import shutil
+        import pathlib
+
+        # Check that the LUT directory exists
+        pathlib.Path(self.lut_dir).mkdir(parents=True, exist_ok=True)
+
+        logger.info("Download AHI LUTs files and store in directory %s",
+                    self.lut_dir)
+        tempdir = tempfile.gettempdir()
+        fname = os.path.join(tempdir, 'tmp.tgz')
+        # Download the LUTs
+        self._download_luts(fname)
+
+        # The file is tarred, untar and remove the downloaded file
+        self._untar_luts(fname, tempdir)
+
+        lut_dl_dir = os.path.join(tempdir, 'count2tbb/')
+
+        # Loop over the LUTs and copy to the correct location
+        for lutfile in AHI_LUT_NAMES:
+            shutil.move(os.path.join(lut_dl_dir, lutfile), os.path.join(self.lut_dir, lutfile))
+        shutil.rmtree(lut_dl_dir)
+
+    def get_dataset(self, key, info):
+        """Get the dataset."""
+        return self.read_band(key, info)
+
+    def get_area_def(self, dsid):
+        """Get the area definition.
+
+        This is fixed, but not defined in the file. So we must
+        generate it ourselves with some assumptions."""
+        if self.areaname == 'fld':
+            area_extent = AHI_FULLDISK_EXTENT
+        else:
+            raise NotImplementedError("Reader only supports full disk data.")
+
+        proj_param = 'EPSG:4326'
+
+        area = geometry.AreaDefinition('gridded_himawari',
+                                       'A gridded Himawari area',
+                                       'longlat',
+                                       proj_param,
+                                       self.ncols,
+                                       self.nlines,
+                                       area_extent)
+        self.area = area
+
+        return area
+
+    def _read_data(self, fp_):
+        """Read raw binary data from file."""
+        return da.from_array(np.memmap(self.filename,
+                                       offset=fp_.tell(),
+                                       dtype='>u2',
+                                       shape=(self.nlines, self.ncols),
+                                       mode='r'),
+                             chunks=CHUNK_SIZE)
+
+    def read_band(self, key, info):
+        """Read the data."""
+        tic = datetime.now()
+
+        with open(self.filename, "rb") as fp_:
+            res = self._read_data(fp_)
+        logger.debug("Reading time " + str(datetime.now() - tic))
```

Comment:
I know this is present in multiple reader, but I think it should be phased out, since the lazy reading makes this irrelevant...